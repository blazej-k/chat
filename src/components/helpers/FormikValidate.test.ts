import formikValidate from './FormikValidate'

describe('Formik validate', () => {
    test('should return object with login error', () => {
        const values: FormValues = {login: '', password: 'q', keepLogIn: true}
        const res = formikValidate(values, 'signin')
        expect(res.login).toBeTruthy()
    })
    test('should return object with password error', () => {
        const values: FormValues = {login: 'q', password: '', keepLogIn: true}
        const res = formikValidate(values, 'signin')
        expect(res.password).toBeTruthy()
    })
    test('should return object with sex error', () => {
        const values: FormValues = {login: 'q', password: '', keepLogIn: true}
        const res = formikValidate(values, 'signup')
        expect(res.sex).toBeTruthy()
    })
    test('should return empty errors object(signin)', () => {
        const values: FormValues = {login: 'q', password: 'q', keepLogIn: true}
        const res = formikValidate(values, 'signin')
        expect(res).toEqual({})
    })
    test('should return empty errors object(signup)', () => {
        const values: FormValues = {login: 'q', password: 'q', keepLogIn: true, sex: 'female'}
        const res = formikValidate(values, 'signup')
        expect(res).toEqual({})
    })
})