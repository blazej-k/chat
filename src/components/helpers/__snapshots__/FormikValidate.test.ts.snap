// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Formik validate should return object with login error 1`] = `true`;

exports[`Formik validate should return object with password error 1`] = `true`;

exports[`Formik validate should return object with sex error 1`] = `true`;
